### PR TITLE
Bug fix: Add Validators for prompts to not allow bad data into context and login commands

### PR DESCRIPTION
**Pull request by mpanchajanya**
### What this PR does / why we need it

-  Added required and no empty spaces validators to prompts in tanzu context and tanzu login commands to avoid having bad data in the config yaml files.

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

Below testing show Required and no only spaces validators in action

```
mpanchajanya@mpanchajanF09MK ~/v/r/tanzu-cli (bug-fix-do-not-allow-empty-context-names)> tanzu context create
? Select context creation type Control plane endpoint
X Sorry, your reply was invalid: value contains only spaces
? Enter control plane endpoint test
X Sorry, your reply was invalid: value contains only spaces
? Give the context a name test
[i] Failed to test for vSphere supervisor: Get "test/wcp/loginbanner": unsupported protocol scheme ""
[x] : error creating kubeconfig with tanzu pinniped-auth login plugin: Get "test/wcp/loginbanner": unsupported protocol scheme ""
[x] : error creating kubeconfig with tanzu pinniped-auth login plugin: Get "test/wcp/loginbanner": unsupported protocol scheme ""
mpanchajanya@mpanchajanF09MK ~/v/r/tanzu-cli (bug-fix-do-not-allow-empty-context-names) [1]> tanzu context create
? Select context creation type Local kubeconfig
X Sorry, your reply was invalid: Value is required
? Enter path to kubeconfig (if any) test
X Sorry, your reply was invalid: Value is required
? Enter kube context to use test
X Sorry, your reply was invalid: Value is required
? Give the context a name test
```

```
mpanchajanya@mpanchajanF09MK tanzu-cli % tanzu login  
Command "login" is deprecated, will be removed in version "1.2.0". Use "context" instead.
? Select login type Server endpoint
X Sorry, your reply was invalid: Value is required
? Enter server endpoint test
X Sorry, your reply was invalid: Value is required
? Give the server a name test
[i] Failed to test for vSphere supervisor: Get "test/wcp/loginbanner": unsupported protocol scheme ""
[x] : error creating kubeconfig with tanzu pinniped-auth login plugin: Get "test/wcp/loginbanner": unsupported protocol scheme ""
[x] : error creating kubeconfig with tanzu pinniped-auth login plugin: Get "test/wcp/loginbanner": unsupported protocol scheme ""
mpanchajanya@mpanchajanF09MK tanzu-cli % 


```

### Release note

```release-note
Added validators to tanzu context and tanzu login to prevent bad data to persist into config
```

### CODE REVIEW

File: pkg/command/ceip_participation.go [hunk 1]
```diff
@@ -53,10 +53,10 @@ func newCEIPParticipationSetCmd() *cobra.Command {
 		Long:  "Set the opt-in preference for CEIP (subject to change)",
 		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if !strings.EqualFold(args[0], "true") && !strings.EqualFold(args[0], "false") {
+			if !strings.EqualFold(args[0], True) && !strings.EqualFold(args[0], "false") {
```
Comment (vuil) [inline pkg/command/ceip_participation.go:56]: Can you explain what the point of the change is? How was it causing problems, and why is handling of false and true different?

Comment (mpanchajanya) [reply pkg/command/ceip_participation.go]: Added True and False const

Comment (mpanchajanya) [reply pkg/command/ceip_participation.go]: Updated to use the strings.EqualFold()

**Review by vuil — APPROVED** [review]
lgtm, thanks!

**Pull request merged**